packages/lodestar/src/eth1/impl/ethers.ts

```diff
@@ -143,79 +161,81 @@ export class EthersEth1Notifier extends (EventEmitter as { new(): Eth1EventEmitt
+      this.db.depositData.batchPut(blockDepositEvents.map((depositEvent) => ({
         key: depositEvent.index,
         value: depositEvent,
       }))),
       // deposit data roots
-      this.db.depositDataRoot.batchPut(depositEvents.map((depositEvent) => ({
+      this.db.depositDataRoot.batchPut(blockDepositEvents.map((depositEvent) => ({
         key: depositEvent.index,
         value: this.config.types.DepositData.hashTreeRoot(depositEvent),
       }))),
     ]);
-    const depositTree = await this.db.depositDataRoot.getTreeBacked(this.lastProcessedDepositIndex);
+    const depositCount = blockDepositEvents[blockDepositEvents.length - 1].index + 1;
+    if (depositCount >= this.config.params.MIN_GENESIS_ACTIVE_VALIDATOR_COUNT) {
+      return await this.processEth1Votes(blockNumber, blockDepositEvents);
+    }
+    return true;
+  }
+
+  /**
+   * Process proposing data of eth1 block
+   * @param blockNumber
+   * @param blockDepositEvents
+   * @returns true if success
+   */
+  public async processEth1Votes(blockNumber: number, blockDepositEvents: IDepositEvent[]): Promise<boolean> {
+    this.logger.verbose(`Processing proposing data of eth1 block ${blockNumber}`);
+    const block = await this.getBlock(blockNumber);
+
+    if (!block) {
+      this.logger.verbose(`eth1 block ${blockNumber} not found`);
+      return false;
+    }
+    const depositTree = await this.db.depositDataRoot.getTreeBacked(blockDepositEvents[0].index - 1);
+    const depositCount = blockDepositEvents[blockDepositEvents.length - 1].index + 1;
     const eth1Data = {
       blockHash: fromHexString(block.hash),
       depositRoot: depositTree.tree().root,
-      depositCount: this.lastProcessedDepositIndex + 1,
+      depositCount,
     };
     // eth1 data
     await this.db.eth1Data.put(block.timestamp, eth1Data);
-    this.lastProcessedEth1BlockNumber++;
+    this.lastProcessedEth1BlockNumber = blockNumber;
     // emit events
-    depositEvents.forEach((depositEvent) => {
+    blockDepositEvents.forEach((depositEvent) => {
       this.emit("deposit", depositEvent.index, depositEvent);
     });
     this.emit("eth1Data", block.timestamp, eth1Data, blockNumber);
-    this.processingBlock = false;
     return true;
   }
 
-  public async getDepositEvents(blockTag: string | number): Promise<IDepositEvent[]> {
+  public async getDepositEvents(fromBlockTag: string | number, toBLockTag?: string | number): Promise<IDepositEvent[]> {
     return (await this.provider.getLogs({
-      fromBlock: blockTag,
-      toBlock: blockTag,
+      fromBlock: fromBlockTag,
+      toBlock: toBLockTag || fromBlockTag,
       address: this.contract.address,
       topics: [this.contract.interface.events.DepositEvent.topic],
     })).map((log) => {
-      return this.parseDepositEvent(this.contract.interface.parseLog(log).values);
+      const blockNumber = log.blockNumber;
+      return this.parseDepositEvent(this.contract.interface.parseLog(log).values, blockNumber);
```

Comment:
You might consider putting the `this.contract.interface.parseLog` call inside `parseDepositEvent`, and only passing in the single arg `log`